### DISCUSSION TITLE
Using conlist and constr show type Any in type intellisense

### DISCUSSION THREAD

**Discussion by IDrumsey** [pydantic/pydantic#7047] [Question] [answered]
```python
from pydantic import BaseModel, conlist, constr


class MyValidator(BaseModel):
    testProp: conlist(str, min_length=1)


myData = MyValidator(**{"testProp": ['asdf1', 'asdf2']})

print(type(myData.testProp))
```

Using this code in vscode, when I hover over the `myData.testProp`, it shows it as `Any` type. However if I print the type using `print(type(myData.testProp))`, it prints `<class 'list'>`. Anybody else having this issue? Not sure if it's a me problem or a Pydantic problem. Using version 2, but I think this was an issue when I was using version 1 as well.

This might be applying to all the con* functions, not sure.

**Comment by Viicos** [comment, marked as answer]
This is probably due to the fact that pyright doesn't accept function calls in annotations:

`Call expression not allowed in type expression` (with pylance strict).

A workaround would be to explicitly use the returned annotation by `conlist`:

```python
from typing import Annotated

import annotated_types
from pydantic import BaseModel, conlist, constr


class MyValidator(BaseModel):
    testProp: Annotated[list[str], annotated_types.Len(min_length=1)]
```

Edit: the `con*` functions are going to be deprecated in V3. Instead, refer to the sections of each type [here](https://docs.pydantic.dev/latest/api/standard_library_types/) to figure out how to apply constraints.

**Reply by akshay-acog** [reply]
Nice
